python/pyarrow/tests/test_dataset.py

```diff
@@ -3011,7 +3012,8 @@ def _create_parquet_dataset_simple(root_path):
     for i in range(4):
         table = pa.table({'f1': [i] * 10, 'f2': np.random.randn(10)})
         pq.write_to_dataset(
-            table, str(root_path), metadata_collector=metadata_collector
+            table, str(root_path), use_legacy_dataset=True,
+            metadata_collector=metadata_collector
```

Comment:
Was there a reason for specifying True here specifically? (`metadata_collector` should be supported with `use_legacy_dataset=False` as well)

Reply:
Not sure right now, changing to `use_legacy_dataset=False` seems to be working 👍